Core/PARStore.m

```diff
@@ -1398,9 +1407,15 @@ - (void)applySyncChangeWithValues:(NSDictionary *)values timestamps:(NSDictionar
     NSAssert([self.memoryQueue isInCurrentQueueStack], @"%@:%@ should only be called from within the memory queue", [self class],NSStringFromSelector(_cmd));
     if (self._inMemoryCacheEnabled)
     {
-        [values enumerateKeysAndObjectsUsingBlock:^(id key, id obj, BOOL *s)     { self._memory[key] = obj;              }];
+        [values enumerateKeysAndObjectsUsingBlock:^(id key, id obj, BOOL *s)
+        {
+            self._memory[key] = (obj != [NSNull null] ? obj : nil);
```

Comment:
So if the obj is Null, we remove the key from the memory dictionary? Isn't that a problem for whatever is using the PARStore? Doesn't it need to know the difference between something that has been set to nil explicitly, and something which has never been set? Don't we have to carry that NSNull through the whole system in some form? In the case of Agenda, my guess is that NSNull would become an AnyObject? with a nil value.

Reply:
Conclusion from our Skype call is that this is the desired behaviour

Reply:
Addressed